mqueue/rtqueue/rtqueue.go

```diff
@@ -0,0 +1,72 @@
+package rtqueue
+
+import (
+	"go.uber.org/atomic"
+
+	"github.com/onflow/flow-go/mqueue"
+
+	"github.com/onflow/flow-go/utils/fifoqueue"
+)
+
+// Queue is a real-time queue which guarantees that items added to the queue
+// will immediately be available to be received over Recv channel.
+type Queue interface {
+	Add(item mqueue.Message)
+
+	Recv() <-chan mqueue.Message
+}
+
+type queue struct {
+	items      *fifoqueue.FifoQueue
+	in, out    chan mqueue.Message
+	shovelling *atomic.Bool
+}
+
+func New() Queue {
+	fifo, _ := fifoqueue.NewFifoQueue()
+	q := &queue{
+		items:      fifo,
+		in:         make(chan mqueue.Message),
+		out:        make(chan mqueue.Message),
+		shovelling: atomic.NewBool(false),
+	}
+	return q
+}
+
+// shovel is a goroutine that lives for as long as the queue is non-empty.
+func (q *queue) shovel() {
+
+	// shovel is always started when at least one item is being added
+	head := <-q.in
```

Comment:
Sorry, I might be completely misunderstanding. Here my thoughts:
* you are directly reading from the tail channel here
* so if you have elements in the `items`: won't this tail element will jump the queue (?) https://github.com/onflow/flow-go/blob/2f8380cdae6c1f709179b24766eac884e5b6eb7c/mqueue/rtqueue/rtqueue.go#L48
* I think you would need to first check `items`, grep the head from `items` if there is any. Only if `items` is empty, you can read from `q.in`